### ISSUE TITLE
sessionVariables reset between scenarios

### ISSUE THREAD

**Issue by asvdw** [serenity-bdd/serenity-core#1174]
I think there is a problem with sessionVariables with serenity 1.9.x (I'm using serenity-cucumber): session Variables are not shared between scenarios.

Ok with serenity 1.8 / cucumber 1.6

*Step to reproduce*
* In a scenario, use this command:
```
     Serenity.setSessionVariable(key).to(value);
```
* On another scenario:
```
        assert Serenity.hasASessionVariableCalled(key) ;
        assert Serenity.sessionVariableCalled(key).equals(value);
```

*Output*
* java.lang.AssertionError: No var with this key 

*Sample project*
https://github.com/asvdw/serenity/tree/sessionVariables
on branch sessionVariables, all tests pass
Now change pom.xml to set serenity version to 1.9.x
--> error 

Unfortunately I have no time right now to look further so I'm now using a simple Singleton to store my session variables.

**Comment by wakaleo** [comment]
This is expected behaviour - session variables should be reset between scenarios, as scenarios should be independent.

**Comment by asvdw** [comment]
That was not the case with 1.8 but ok, I understand.
IMHO in real life scenarios can't _always_ be fully independent. Yes I know, that's not the right way, I won't discuss why I choose that on one project but it saves be lots of time and headaches.

So sessionVariables are only used to share data between classes on same scenario.

Ok, thank you for your reply, I'l go on with my "bad" singleton...

**Comment by wakaleo** [comment]
It was actually a bug raised and fixed in 1.9 -  it’s hard to satisfy all the use cases.
